examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1405 @@
+    };
+
+    // Parse the parameter.
+    let params: OnReceivingCis2Params<ContractTokenId, ContractTokenAmount> =
+        ctx.parameter_cursor().get()?;
+
+    // Build the transfer from this contract to the contract owner.
+    let transfer = Transfer {
+        token_id: params.token_id,
+        amount:   params.amount,
+        from:     Address::Contract(ctx.self_address()),
+        to:       Receiver::from_account(ctx.owner()),
+        data:     AdditionalData::empty(),
+    };
+
+    let parameter = TransferParams::from(vec![transfer]);
+
+    // Send back a transfer
+    host.invoke_contract_read_only(
+        &sender,
+        &parameter,
+        EntrypointName::new("transfer")?,
+        Amount::zero(),
+    )?;
+    Ok(())
+}
+
+/// Get the supported standards or addresses for a implementation given list of
+/// standard identifiers.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+#[receive(
+    contract = "cis2_multi_royalties",
+    name = "supports",
+    parameter = "SupportsQueryParams",
+    return_value = "SupportsQueryResponse",
+    error = "ContractError"
+)]
+fn contract_supports<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<SupportsQueryResponse> {
+    // Parse the parameter.
+    let params: SupportsQueryParams = ctx.parameter_cursor().get()?;
+
+    // Build the response.
+    let mut response = Vec::with_capacity(params.queries.len());
+    for std_id in params.queries {
+        if SUPPORTS_STANDARDS.contains(&std_id.as_standard_identifier()) {
+            response.push(SupportResult::Support);
+        } else {
+            response.push(host.state().have_implementors(&std_id));
+        }
+    }
+    let result = SupportsQueryResponse::from(response);
+    Ok(result)
+}
+
+/// Set the addresses for an implementation given a standard identifier and a
+/// list of contract addresses.
+///
+/// It rejects if:
+/// - Sender is not the owner of the contract instance.
+/// - It fails to parse the parameter.
+#[receive(
+    contract = "cis2_multi_royalties",
+    name = "setImplementors",
+    parameter = "SetImplementorsParams",
+    error = "ContractError",
+    mutable
+)]
+fn contract_set_implementor<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    // Authorize the sender.
+    ensure!(ctx.sender().matches_account(&ctx.owner()), ContractError::Unauthorized);
+    // Parse the parameter.
+    let params: SetImplementorsParams = ctx.parameter_cursor().get()?;
+    // Update the implementors in the state
+    host.state_mut().set_implementors(params.id, params.implementors);
+    Ok(())
+}
+
+/// Returns the address of the payee and the amount of royalties to pay them
+/// 
+/// It rejects if:
+/// - The contract does not pay royalties
+/// - The return result would be 0
+/// - The token is not recognised
```

Comment:
```suggestion
/// - The token id does not exist in state
```

Reply:
@NossicaSystems Please address or comment if you want to keep the current text.

Reply:
Updated as suggested